ci/scrape-targets.py

```diff
@@ -0,0 +1,16 @@
+from bs4 import BeautifulSoup
+import requests
+
+url = "https://doc.rust-lang.org/nightly/rustc/platform-support.html"
```

Comment:
Is there a way we can get this from the installed rustc rather than an online source?

Reply:
It looks like we can install the docs and scrape a local HTML file? Trying this, it's a bit fiddly to test locally because `./miri toolchain -c rust-docs` doesn't add them to the toolchain, but if the `miri` toolchain doesn't exist yet it looks like the component is installed.

Reply:
Ah... well if it's complicated we can also stick to the URL and hope we don't get spurious failures from that.

Reply:
> ./miri toolchain -c rust-docs doesn't add them to the toolchain

So this worked after all?

Reply:
The CI job works because there is no pre-existing `miri` toolchain when we run `./miri toolchain -c rust-docs`. It's functional, but awkward to test manually.

Reply:
Ah that's what you mean.

I am not sure if RTIM supports adding components to an already installed toolchain.

Reply:
Apart from scraping platform-support.html you could also look in https://static.rust-lang.org/dist/channel-rust-nightly.toml to find all targets for which we ship libstd (all tier 1 and tier 2 targets) In case of a rustup installed toolchain it is available locally under `$(rustc --print sysroot)/lib/rustlib/multirust-channel-manifest.toml`. This doesn't distinguish between tier 1 and tier 2 though, but you could distinguish between with and without host tools by looking if rustc is available for the target or not.

Reply:
Or we just test tier 1 as well, it should only take a few extra minutes.

That would probably be more reliable than scraping html, yeah. Not a blocker to landing the initial experiment though.